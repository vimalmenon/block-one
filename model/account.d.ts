export interface AccountParams {
  customerId: string
  accountNo: string
}

export interface AccountModel {
  customerId: number
  accountNumber: string
  accountType: string
  balance: number
}
