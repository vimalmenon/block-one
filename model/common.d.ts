export type GenericMethodType<K> = (value: K) => void
export type GenericReturnMethodType<K, V> = (value: K) => V

export interface TableColumn {
  header: string
  accessor: string
  template?: any
}
