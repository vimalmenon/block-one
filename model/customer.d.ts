export interface CustomerParams {
  customerId: string
}

export interface CustomerModel {
  prefix: string
  name: string
  customerId: number
  address: string
  emailAddress: string
  contactNo: string
}
