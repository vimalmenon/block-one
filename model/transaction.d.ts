export interface TransactionModel {
  accountNumber: string
  timestamp: string
  action: string
  description: string
  amount: number
  currency: 'HKD' | 'USD'
}
