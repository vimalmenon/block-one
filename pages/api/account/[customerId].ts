import type { NextApiRequest, NextApiResponse } from 'next';
import {AccountModel} from "../../../model/account";

import {openDb} from "../../../database.setup";

export default async (req: NextApiRequest, res: NextApiResponse<AccountModel[]>) => {
    if (req.method ==="GET") {
        const {customerId, transfer} = req.query;
        let result;
        if (transfer) {
            result = await (await openDb()).all('SELECT customer_id as customerId, account_number as accountNumber, account_type as accountType, balance FROM accounts WHERE customer_id = ? and account_type IN ("Saving Account", "Checking Account")', customerId);
        } else {
            result = await (await openDb()).all('SELECT customer_id as customerId, account_number as accountNumber, account_type as accountType, balance FROM accounts WHERE customer_id = ?', customerId);
        }

        res.status(200).json(result);
    }
}