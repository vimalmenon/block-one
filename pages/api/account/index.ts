import type { NextApiRequest, NextApiResponse } from 'next';
import {AccountModel} from "../../../model/account";
import {CustomerModel} from "../../../model/customer"

//Last Login


const customerData:CustomerModel[] = [
    {
        prefix:"Mrs",
        name:"Jane Doe",
        customerId: 12345,
        address:"Kowloon, HK",
        emailAddress:"jane.doe@example.com",
        contactNo:"1234 7889"
    },
    {
        prefix:"Mr",
        name:"John Doe",
        customerId: 67890,
        address:"Central, HK",
        emailAddress:"John.doe@example.com",
        contactNo:"7889 1234"
    }
];


const customerAccountDetails:AccountModel[] = [
    {
        customerId:12345,
        accountNumber: "123-456-789",
        accountType: "Saving Account",
        balance:20000
    },
    {
        customerId:12345,
        accountNumber: "123-456-788",
        accountType: "Checking Account",
        balance:10000
    },
    {
        customerId:12345,
        accountNumber: "123-456-788",
        accountType: "Current Account",
        balance:5000
    },
    {
        customerId:12345,
        accountNumber: "123-456-787",
        accountType: "MPF Account",
        balance:2000
    },
    {
        customerId:67890,
        accountNumber: "123-456-786",
        accountType: "Saving Account",
        balance:40000
    },
    {
        customerId:67890,
        accountNumber: "123-456-785",
        accountType: "Checking Account",
        balance:30000
    },
    {
        customerId:67890,
        accountNumber: "123-456-788",
        accountType: "Current Account",
        balance:8000
    },
    {
        customerId:67890,
        accountNumber: "123-456-784",
        accountType: "MPF Account",
        balance:3000
    }
]

export default (req: NextApiRequest, res: NextApiResponse<any>) => {
    if (req.method ==="GET") {
        const {accountId} = req.query;
        const selectedCustomer = customerAccountDetails.filter((customer) => {
            return customer.customerId.toString() === accountId;
        })
        res.status(200).json(selectedCustomer);
    }
  
}