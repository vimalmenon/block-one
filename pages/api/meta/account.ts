import type { NextApiRequest, NextApiResponse } from 'next';
import {AccountParams} from "../../../model/account";

import {openDb} from "../../../database.setup";

export default async (req: NextApiRequest, res: NextApiResponse<AccountParams[]>) => {
  if (req.method==="GET") {
    const results = await (await openDb()).all("select customer_id as customerId,account_number as accountNo  from accounts");
    const params:AccountParams[] = []
    results.forEach((result) => {
      params.push({
        customerId: result.customerId.toString(),
        accountNo: result.accountNo
      });
    });
    res.status(200).json(params)
  }
}