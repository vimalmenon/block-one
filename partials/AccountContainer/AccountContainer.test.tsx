import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AccountModel } from '../../model/account'

import { AccountContainer } from './'

describe('AccountContainer Testing', () => {
  test('AccountContainer : Snapshot testing', () => {
    const onAccountSelect = jest.fn()
    const accounts: AccountModel[] = [
      {
        customerId: 1234,
        accountNumber: '12233344',
        accountType: 'Saving Account',
        balance: 20000,
      },
    ]
    const wrapper = shallow(
      <AccountContainer
        title="From Account"
        onAccountSelect={onAccountSelect}
        accounts={accounts}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('AccountContainer : Snapshot testing with selectedAccount', () => {
    const onAccountSelect = jest.fn()
    const accounts: AccountModel[] = [
      {
        customerId: 1234,
        accountNumber: '12233344',
        accountType: 'Saving Account',
        balance: 20000,
      },
    ]
    const selectedAccount: AccountModel = {
      customerId: 1234,
      accountNumber: '12233344',
      accountType: 'Saving Account',
      balance: 20000,
    }
    const wrapper = shallow(
      <AccountContainer
        title="From Account"
        onAccountSelect={onAccountSelect}
        accounts={accounts}
        selectedAccount={selectedAccount}
        isFromAccount={true}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('AccountContainer : Snapshot testing with selectedAccount and transferAmount', () => {
    const onAccountSelect = jest.fn()
    const accounts: AccountModel[] = [
      {
        customerId: 1234,
        accountNumber: '12233344',
        accountType: 'Saving Account',
        balance: 20000,
      },
    ]
    const selectedAccount: AccountModel = {
      customerId: 1234,
      accountNumber: '12233344',
      accountType: 'Saving Account',
      balance: 20000,
    }
    const wrapper = shallow(
      <AccountContainer
        title="From Account"
        onAccountSelect={onAccountSelect}
        accounts={accounts}
        selectedAccount={selectedAccount}
        isFromAccount={false}
        transferAmount={2000}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  test('AccountContainer : Snapshot testing Select change event', () => {
    const onAccountSelect = jest.fn()
    const accounts: AccountModel[] = [
      {
        customerId: 1234,
        accountNumber: '12233344',
        accountType: 'Saving Account',
        balance: 20000,
      },
    ]
    const selectedAccount: AccountModel = {
      customerId: 1234,
      accountNumber: '12233344',
      accountType: 'Saving Account',
      balance: 20000,
    }
    const wrapper = shallow(
      <AccountContainer
        title="From Account"
        onAccountSelect={onAccountSelect}
        accounts={accounts}
        selectedAccount={selectedAccount}
        isFromAccount={false}
        transferAmount={2000}
      />
    )
    wrapper.find('Select').simulate('change', { key: 'Change', target: { value: 'mock' } })
    expect(onAccountSelect).toHaveBeenCalled()
    expect(onAccountSelect).toHaveBeenCalledWith('mock')
  })
})
