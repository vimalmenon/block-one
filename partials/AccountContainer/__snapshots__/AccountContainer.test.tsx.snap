// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountContainer Testing AccountContainer : Snapshot testing 1`] = `
<div
  className="flex w-5/12 flex-col border shadow"
>
  <div
    className="flex h-20 text-3xl items-center justify-center"
  >
    From Account
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Select an account
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        <Select
          items={
            Array [
              Object {
                "accountNumber": "12233344",
                "accountType": "Saving Account",
                "balance": 20000,
                "customerId": 1234,
              },
            ]
          }
          nameProps="accountNumber"
          onChange={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AccountContainer Testing AccountContainer : Snapshot testing with selectedAccount 1`] = `
<div
  className="flex w-5/12 flex-col border shadow"
>
  <div
    className="flex h-20 text-3xl items-center justify-center"
  >
    From Account
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Select an account
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        <Select
          items={
            Array [
              Object {
                "accountNumber": "12233344",
                "accountType": "Saving Account",
                "balance": 20000,
                "customerId": 1234,
              },
            ]
          }
          nameProps="accountNumber"
          onChange={[Function]}
          value="{\\"customerId\\":1234,\\"accountNumber\\":\\"12233344\\",\\"accountType\\":\\"Saving Account\\",\\"balance\\":20000}"
        />
      </div>
    </div>
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Account details
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        Saving Account
      </div>
    </div>
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Account balance
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        20,000
      </div>
    </div>
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Amount
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        <Input
          error={false}
          onChange={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AccountContainer Testing AccountContainer : Snapshot testing with selectedAccount and transferAmount 1`] = `
<div
  className="flex w-5/12 flex-col border shadow"
>
  <div
    className="flex h-20 text-3xl items-center justify-center"
  >
    From Account
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Select an account
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        <Select
          items={
            Array [
              Object {
                "accountNumber": "12233344",
                "accountType": "Saving Account",
                "balance": 20000,
                "customerId": 1234,
              },
            ]
          }
          nameProps="accountNumber"
          onChange={[Function]}
          value="{\\"customerId\\":1234,\\"accountNumber\\":\\"12233344\\",\\"accountType\\":\\"Saving Account\\",\\"balance\\":20000}"
        />
      </div>
    </div>
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Account details
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        Saving Account
      </div>
    </div>
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Account balance
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        20,000
      </div>
    </div>
    <div
      className="w-full flex p-4"
    >
      <div
        className="flex w-1/3"
      >
        <Label>
          Updated balance
        </Label>
      </div>
      <div
        className="flex w-2/3"
      >
        22000
      </div>
    </div>
  </div>
</div>
`;
